### DISCUSSION TITLE
How to run two machine learning models in one CV application?

### DISCUSSION THREAD

**Discussion by ThenoobMario** [AxisCommunications/acap-computer-vision-sdk-examples#184] [Q&A] [answered]
Taking inspiration from [object-detector-python](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/main/object-detector-python), can I run two tflite models in the same CV applications? If yes, how?

One way I think it can be done is - Run two inference-server containers with corresponding models and call them. But I am not sure if I need to define two unix sockets for the same. Is there a better alternative?

**Comment by Corallo** [comment, marked as answer]
Hi, yes it is possible.

You would have to add another -m flag here:
https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/blob/main/object-detector-python/docker-compose.yml#L28
so that it looks like:
`entrypoint: ["/opt/app/acap_runtime/acapruntime", "-m", "${MODEL_PATH}", "-m", "${MODEL_PATH_2}", "-j", "${INFERENCE_CHIP}"]`
You define the new model path here:
https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/blob/main/object-detector-python/config/env.aarch64.artpec8
And add the model to this docker file:
https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/blob/main/object-detector-python/Dockerfile.model

Then, in your application, you should just be able to do two calls like this:
https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/blob/main/object-detector-python/app/detector.py#L40
with the two different model paths

**Reply by ThenoobMario** [reply]
Hi @Corallo,

Thank you! This worked!

As a separate question, is it possible to define different chips as well that run two different things?

**Reply by Corallo** [reply]
No, the ACAP runtime does not support running on multiple chips.
The ACAP runtime is open source, you could implement this feature yourself: https://github.com/AxisCommunications/acap-runtime/tree/main, but is probably a lot of work.
I would rather recommend making a native ACAP at that point instead, to have more control on the chip used for each model:
https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/object-detection
Which gives you other benefits like GUI support, and better performance

**Reply by ThenoobMario** [reply]
Alright, makes sense. Thanks for the quick response.
